File: 0001_0100/0010/0010.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    a, b, c, d = (int(x) for x in f.read().split())
    q.write(' '.join(str(x) for x in range(-100, 101) if a * x ** 3 + b * x ** 2 + c * x + d == 0))

File: 0001_0100/0010/0010.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var a, b, c, d, i int64
	fmt.Fscan(f, &a, &b, &c, &d)
	for i = -100; i <= 100; i++ {
		if a*i*i*i+b*i*i+c*i+d == 0 {
			fmt.Fprint(q, i, " ")
		}
	}
}
